spec/version_requirement_spec.rb

```diff
@@ -0,0 +1,28 @@
+describe MetadataJsonLint::VersionRequirement do
+  describe '.new' do
+    it { expect(described_class.new('')).to be_a(MetadataJsonLint::VersionRequirement) }
+    it { expect(described_class.new('>= 1.0')).to be_a(MetadataJsonLint::VersionRequirement) }
+    it { expect { described_class.new('## 1.0') }.to raise_error(ArgumentError) }
+  end
+
+  describe '#mixed_syntax?' do
+    it { expect(described_class.new('>= 1.0.0').mixed_syntax?).to be false }
+    it { expect(described_class.new('1.0.x').mixed_syntax?).to be false }
```

Comment:
It only uses `.x` - if it was mixed syntax then it'd also be using `>=` or similar.

`.x` by itself is reasonably clear, i.e. 1.0.0 through to 1.0.999, but adding a second operator such as `>= 1.0.x` is what it tries to warn against, as the intent is less clear.